### PR TITLE
fix: Make stack outputs description optional

### PR DESCRIPTION
**Pull request by mildaniel**
#### Which issue(s) does this change fix?

#4658

#### Why is this change necessary?
Don't assume CFN stack outputs will have a description since it's optional.

#### How does it address the issue?
Use dictionary `.get()` method with default value where applicable to avoid getting a `KeyError` when a value doesn't exist.

#### What side effects does this change have?
N/A

#### Mandatory Checklist
**PRs will only be reviewed after checklist is complete**

- [ ] Add input/output [type hints](https://docs.python.org/3/library/typing.html) to new functions/methods
- [ ] Write design document if needed ([Do I need to write a design document?](https://github.com/aws/aws-sam-cli/blob/develop/DEVELOPMENT_GUIDE.md#design-document))
- [ ] Write/update unit tests
- [x] Write/update integration tests
- [ ] Write/update functional tests if needed
- [x] `make pr` passes
- [ ] `make update-reproducible-reqs` if dependencies were changed
- [ ] Write documentation

By submitting this pull request, I confirm that my contribution is made under the terms of the [Apache 2.0 license](https://www.apache.org/licenses/LICENSE-2.0).

### CODE REVIEW

**Review by jfuss — APPROVED** [review]

File: samcli/lib/list/resources/resource_mapping_producer.py [hunk 1]
```diff
@@ -132,11 +132,11 @@ def produce(self):
...
             for deployed_resource in response["StackResources"]:
                 resource_data = ResourcesDef(
-                    LogicalResourceId=deployed_resource["LogicalResourceId"],
-                    PhysicalResourceId=deployed_resource["PhysicalResourceId"],
+                    LogicalResourceId=deployed_resource.get("LogicalResourceId"),
+                    PhysicalResourceId=deployed_resource.get("PhysicalResourceId"),
                 )
                 resources_list.append(dataclasses.asdict(resource_data))
-                seen_resources.add(deployed_resource["LogicalResourceId"])
+                seen_resources.add(deployed_resource.get("LogicalResourceId"))
```
Comment (hnnasit) [inline samcli/lib/list/resources/resource_mapping_producer.py:139]: If `deployed_resource.get("LogicalResourceId")` returns None, None gets added to the seen_resources set. Should we avoid adding None here and check it with an if condition instead?

**Review by hnnasit — CHANGES_REQUESTED** [review]

File: samcli/lib/list/resources/resource_mapping_producer.py (hunk 1 above)
Comment (mildaniel) [reply samcli/lib/list/resources/resource_mapping_producer.py:139]: We could to be extra safe, but in this case if a resource is already deployed it must have a logical ID and a Physical ID, so we should be okay.

Comment (hnnasit) [reply samcli/lib/list/resources/resource_mapping_producer.py:139]: Right makes sense, in that case should we keep the [] brackets for getting the LogicalResourceId and PhysicalResourceId if these keys are always assumed to be present?

Comment (mildaniel) [reply samcli/lib/list/resources/resource_mapping_producer.py:139]: I do think it's still better practice to use `.get()` as it's a safer operation here. If downstream there's an issue we can deal with it there, but at least we won't be getting a `KeyError` here.

Comment (hnnasit) [reply samcli/lib/list/resources/resource_mapping_producer.py:139]: Fair enough

**Review by hnnasit — APPROVED** [review]

**Pull request merged**